src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -294,21 +294,24 @@ public AzureVMDeploymentInfo createDeployment(
             putVariable(tmp, "cloudTag", template.getAzureCloud().getCloudName());
 
             // add purchase plan for image if needed in reference configuration
-            if (!useCustomImage && StringUtils.isBlank(template.getImageId())) {
-                VirtualMachineImage image = azureClient.virtualMachineImages().getImage(locationName,
-                        template.getImagePublisher(), template.getImageOffer(), template.getImageSku(), template.getImageVersion());
-                if (image != null) {
-                    PurchasePlan plan = image.plan();
-                    if (plan != null) {
-                        ArrayNode resources = (ArrayNode) tmp.get("resources");
-                        for (JsonNode resource : resources) {
-                            String type = resource.get("type").asText();
-                            if (type.contains("virtualMachine")) {
-                                ObjectNode planNode = mapper.createObjectNode();
-                                planNode.put("name", plan.name());
-                                planNode.put("publisher", plan.publisher());
-                                planNode.put("product", plan.product());
-                                ObjectNode.class.cast(resource).replace("plan", planNode);
+            if (!isBasic && !useCustomImage && StringUtils.isBlank(template.getImageId())) {
```

Comment:
what is isBasic?

Reply:
basic image means built-in image, now only support Ubuntu 16.04 LTS and Window Server 2016

Reply:
the combined condition here looks complex, error-prone, hard to read. is useCustomImage,  there should be 2 ways to get Image, one per getImageId(), another one per combination of publisher, offer, sku etc. is it true? is it in this flow?

If isBasic,  and user filled in image info, like sku, publisher, offer etc, what's code flow then?

suggest to refine this condition check, adding more comments

Reply:
There are four ways to provide images here.
isBasic represents built-image, user only have two images to choose from, Unbuntu or Winsever
useCustomImage represents Custom User Image, user have to provide image URI
getImageId() represents Custom Managed Image, user have to provide image ID generated by Packer
the last one is Image Reference, user have to provide  sku, publisher, offer, version

If is Basic, there is no place to fill in image info.

Reply:
@yungezz 
What about we check parameters sku, publisher, offer, version, if they are not blank, then we enter this flow?

Reply:
agree. on code level, regarding ui design, if user input sku, publisher, offer, version, then jenkins should get image based on input. One question is, will Jenkins fill in default values of those parameters?

Reply:
Have done some tests, some rare scenarios may cause problems here.
If users choose Image Reference configuration first, and they want to change for another one. On this condition, the publisher, offer, sku, version info would still be there. So we still have to check the three variables to rule out other configuration types.